src/bartiq/symbolics/sympy_backends.py

```diff
@@ -163,18 +169,31 @@ def define_function(self, expr: T_expr, func_name: str, function: Callable) -> T
         except TypeError:
             return expr
 
-    def is_constant_int(self, expr: T_expr):
+    def is_constant_int(self, expr: TExpr[Expr]):
         """Return True if a given expression represents a constant int and False otherwise."""
         try:
-            int(str(expr))
+            _ = int(str(expr))
             return True
         except ValueError:
             return False
 
-    def serialize(self, expr: T_expr) -> str:
+    def serialize(self, expr: TExpr[Expr]) -> str:
         """Return a textual representation of given expression."""
+        if isinstance(expr, Number):
+            return str(expr)
         return serialize_expression(expr)
 
+    def compare(self, lhs: TExpr[Expr], rhs: TExpr[Expr]) -> ComparisonResult:
+        difference = self.as_expression(lhs - rhs)
+        if not isinstance(difference, Number):
+            difference = difference.expand()
+        if difference == 0:
```

Comment:
What if (for whatever reasons) the difference is `0.0` and not `0` 🙃 ?

Reply:
Although I think in this particular case this cannot happen, I added additional comparison to 0.0 in [0faae4e](https://github.com/PsiQ/bartiq/pull/115/commits/0faae4ed23be183e11a35c0e9b290b925ac43b6d)